### DISCUSSION TITLE
Handle Big Data with Datatable Primereact NextJS

### DISCUSSION THREAD

**Discussion by Zen225** [primefaces/community#856] [PrimeReact] [answered]
Hello

I Want to Handle Big Data / Large Dataset with Datatable Prime React and i Use Next JS

any recommendation Method to this Case ?

Thankyou
Regards

Ryo

**Comment by melloware** [comment, marked as answer]
I am so glad I wrote this as I answered this question a ton of times.  I have an example showing using REST Services to use the `lazy` feature of the datatable so it Sorts, Filters, Paginates etc Just in time.  Running queries to show the bare minimum to represent the table.

Full working example with Code and in memory DB: https://github.com/melloware/quarkus-primereact

In the example I load 500 DB records but you could load 50,000 and it would be the same performance.
